addons/udes_stock/models/stock_picking.py

```diff
@@ -1159,16 +1159,36 @@ def action_confirm(self):
         self.unlink_empty()
         return res
 
+    def find_empty_pickings_to_unlink(self):
+        """ Finds empty pickings to unlink in self, when it is set, otherwise
+            searches for any empty picking.
+        """
+        if self:
+            records = self.filtered(
+                lambda p:
+                    (len(p.move_lines) == 0
+                    and not p.u_mark
+                    and p.is_locked))
+        else:
+            records = self.search([
+                ('u_mark', '=', False),
+                ('is_locked', '=', True),
+            ])
+            records = records.filtered(lambda p: len(p.move_lines) == 0)
+
+        return records
+
     def unlink_empty(self):
         """
-            Delete pickings in self that are empty, locked and cancelled
+            Delete pickings in self that are empty, locked and marked=False.
+            If self is empty, find and delete any picking with the previous
+            criterion.
             This is to prevent us leaving junk data behind when refactoring
         """
-        self.filtered(lambda p:
-                      (len(p.move_lines) == 0
-                       and not p.u_mark
-                       and p.is_locked)).unlink()
-        return self.exists()
+        records = self.find_empty_pickings_to_unlink()
+        records.unlink()
```

Comment:
My only comment here would be to add batching, in case the amount of data is large (e.g. job hasn't been run in a long time). Not a requirement though, since the job is run every 30 minutes by default.

Reply:
Will add thanks.